triemap/src/main/java/tech/pantheon/triemap/EntrySetSpliterator.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * (C) Copyright 2023 PANTHEON.tech, s.r.o. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package tech.pantheon.triemap;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Spliterator;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.UnaryOperator;
+
+final class EntrySetSpliterator<K, V, M extends TrieMap<K, V>> implements Spliterator<Entry<K, V>> {
+
+    static final int SPLIT_THRESHOLD = 1024;
+    private final int characteristics;
+    private final Entry<K, V>[] entries;
+    private final AtomicInteger currentIndex = new AtomicInteger();
+    private final AtomicInteger lastIndex = new AtomicInteger();
+    private final UnaryOperator<Entry<K, V>> entryWrapper;
+
+    EntrySetSpliterator(final M map, final int characteristics,
+            final UnaryOperator<Entry<K, V>> entryWrapper) {
+        this.characteristics = characteristics;
+        this.entryWrapper = entryWrapper;
+        this.entries = toArray(map);
```

Comment:
err... no, definitely not:
create a benchmark: a map of 1M entries, and 10K times do:
- take the entry set
- open a spliterator
- perform a one tryAdvance()

Compare the time that takes with this proposed change and without it